manifests/init.pp

```diff
@@ -484,6 +487,7 @@
   Boolean $yum_gpg_sign_repo_metadata = $pulp::params::yum_gpg_sign_repo_metadata,
   Optional[String] $yum_gpg_key_id = $pulp::params::yum_gpg_key_id,
   Optional[String] $yum_gpg_cmd = $pulp::params::yum_gpg_cmd,
+  Optional[Integer[0,65535]] $yum_remove_old_repodata_count = $pulp::params::yum_remove_old_repodata_count,
```

Comment:
Since this is optional, set the default value in `params.pp` to `undef`.

Reply:
i was trying to make this be a one commit chnage, rather than also having to modify (downstream only) puppet-katello and puppet-foreman_proxy_content and i think if the default is unset then i'd have to do that.  Which is fine, i just want to clarify that is kinda your intent too

Reply:
Drop the `Optional` and set the default value to 0. I think that gets you what you want the easiest.

Reply:
done!  As per our discussion this would allow it to be configured via heira if needed, and set back to the code default of 4